doc/source/guidelines/optional_packages.rst

```diff
@@ -0,0 +1,248 @@
+  up the install and reducing the possibility of dependency conflicts. The
+  trade-off here is any user who wishes to access features which require the
+  optional package will have to install it piecemeal.
+
+If you choose to implement optional packages for your pyansys library, here are
+some helpful best-practices to follow.
+
+
+Implementing Optional Packages in the Build System
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Here's how to implement and use optional requirements for the three most
+popular build systems.
+
+.. tabs::
+
+   .. tab:: poetry
+
+      .. code-block::
+
+         [tool.poetry.extras]
+         all = [
+             "matplotlib",
+             "pyvista",
+             "pyside",
+         ]
+         plotting = [
+             "matplotlib",
+             "pyvista",
+         ]
+         qt = [
+             "pyside",
+         ]
+
+         Install ``package-name`` with the optional ``qt`` packages with:
+
+         .. code-block::
+
+            poetry install --extras "plotting qt"
+
+   .. tab:: flit
+
+      .. code-block::
+
+         [project.optional-dependencies]
+         all = [
+             "matplotlib",
+             "pyvista",
+             "pyside",
+         ]
+         plotting = [
+             "matplotlib",
+             "pyvista",
+         ]
+         qt = [
+             "pyside",
+         ]
+
+         Install ``package-name`` with the optional ``qt`` packages with:
+
+         .. code-block::
+
+            pip install package-name --extras=all
+
+   .. tab:: setuptools
+
+      .. code-block:: python
+
+         from setuptools import setup
+
+         setup(
+             ...
+             extras_require={
+                'all': ['matplotlib', 'pyvista', 'pyside'],
+                'plotting': ['matplotlib', 'pyvista'],
+                 'qt': ['pyside'],
+             },
+             ...
+         )
+
+         Install ``package-name`` with the optional ``qt`` packages with:
+
+         .. code-block::
+
+            pip install package-name[qt]
+
+
+Implementing Optional Libraries in Features
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+One of the best ways to implement an optional dependency is to execute a "lazy
+import" at runtime for the feature in question. For example, if your library
+has an optional dependency on ``matplotlib``, you can implement it with:
+
+.. code:: python
+
+   def plot(x, y):
+       """Plot two numpy arrays.
+
+       Parameters
+       ----------
+       x : numpy.ndarray
+           Numpy array sized (n, ).
+       y : numpy.ndarray
+           Numpy array sized (n, ).
+
+       Notes
+       -----
+       This function requires ``matplotlib``.
+
+       """
+       try:
+           import matplotlib.pyplot as plt
+       except ModuleNotFoundError:  # pragma: no cover
+           raise ModuleNotFoundError(
+               "Please install matplotlib to use this feature with:\n\n"
```

Comment:
```suggestion
               "To use this feature, install Matplotlib with:\n\n"
```

Reply:
I don't think you can use double back ticks in line 141 but tagging within this topic varies a bit.